src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ChannelPricingRepository.php

```diff
@@ -21,8 +21,7 @@ class ChannelPricingRepository extends EntityRepository implements ChannelPricin
     public function findWithDiscountedPrice(): array
     {
         return $this->createQueryBuilder('o')
-            ->andWhere('o.originalPrice IS NOT NULL')
-            ->andWhere('o.originalPrice > o.price')
+            ->andWhere('o.appliedPromotions != \'[]\'')
```

Comment:
Are we sure, this notation will be the same for postgres?

Reply:
As we don't have a build for postgres, I am not 100% sure